akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -148,6 +148,16 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
       event.getClass.getName, seqNr, persistenceId, cause.getMessage)
   }
 
+  /**
+    * Called when the size of the internal stash is exceeds the capacity of the Stash
+    * 
+    * @param message the message which failed to enqueue
+    */
+  protected def onStashOverflow(message: Any): Unit = {
```

Comment:
Users will do the wrong thing, and we get issues, support cases and such, so I think alternative 2 is appropriate.